delta/plugins/composite-views/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/compositeviews/indexing/CompositeSpaces.scala

```diff
@@ -0,0 +1,106 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.compositeviews.indexing
+
+import cats.syntax.all._
+import ch.epfl.bluebrain.nexus.delta.plugins.blazegraph.client.BlazegraphClient
+import ch.epfl.bluebrain.nexus.delta.plugins.blazegraph.indexing.BlazegraphSink
+import ch.epfl.bluebrain.nexus.delta.plugins.compositeviews.indexing.CompositeViewDef.ActiveViewDef
+import ch.epfl.bluebrain.nexus.delta.plugins.compositeviews.model.CompositeViewProjection
+import ch.epfl.bluebrain.nexus.delta.plugins.compositeviews.model.CompositeViewProjection.{ElasticSearchProjection, SparqlProjection}
+import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.client.ElasticSearchClient.Refresh
+import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.client.{ElasticSearchClient, IndexLabel}
+import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.indexing.ElasticSearchSink
+import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.Iri
+import ch.epfl.bluebrain.nexus.delta.rdf.query.SparqlQuery.SparqlConstructQuery
+import ch.epfl.bluebrain.nexus.delta.sdk.model.BaseUri
+import ch.epfl.bluebrain.nexus.delta.sourcing.config.BatchConfig
+import ch.epfl.bluebrain.nexus.delta.sourcing.stream.Operation
+import ch.epfl.bluebrain.nexus.delta.sourcing.stream.Operation.Sink
+import com.typesafe.scalalogging.Logger
+import monix.bio.Task
+
+/**
+  * Defines the pipes, sinks for the indexing progress as well as the init and destroy tasks for a composite view
+  * @param init
+  *   the task to create the different namespaces and indices
+  * @param destroy
+  *   the task to destroy the different namespaces and indices
+  * @param commonSink
+  *   the sink for the common sparql space
+  * @param queryPipe
+  *   the function to create the query pipe from a [[SparqlConstructQuery]]
+  * @param targetSink
+  *   the function to create a sink for a [[CompositeViewProjection]]
+  */
+final case class CompositeSpaces(
```

Comment:
Given the number of namespaces/indices and sinks to instantiate, this allow to separate this logic from the coordinator